### DISCUSSION TITLE
Application Renewals

### DISCUSSION THREAD

**Discussion by Techguyrosh** [AOT-Technologies/forms-flow-ai#1627] [Q&A]
Has anyone cracked the code as yet for an easier way to do application renewals so a client doesn't need to re enter all the information just load last application data and edit?

I have forms that have probably over 100 entry fields, and applications that need to be done every year, so it would help.

**Comment by abhilash-aot** [comment]
We donot have any particular renewal application method at the moment. We can pre load data and fill form fields from any external db or even from older submissions through apis.  This way using this workaround you can pre fill older data May be using the user id  or any particular key in the form to pre fill the data of the last year applications . If you can give bit more brief about the application we can try some samples and let you know. Using the submission or external apis we should be able to achieve this.

**Reply by Techguyrosh** [reply]
Interesting. And would love to try this. So here is a summary of the use case.
I have 14 forms each form has various renewal periods, where persons are required upon the expiry of their previous applications to re submit, The renewal times range from 1 to 4 years maximum. Each Application currently is uniformed with 5 Tabs: A welcome Tab with information on the Application ahead, A Applicant Information Tab, An Entity Tab which gives information on what is being applied for, A Document Upload Tab (This will definately be updated during renewal) and a Declaration Tab (Again will need to be updated). We essentially would have an email prompt users based on their submission dates of the original application to resubmit a renewal if their previous application status was set to approved.

**Reply by Techguyrosh** [reply]
@abhilash-aot any further advise on this one
